src/NHibernate/NHibernateUtil.cs

```diff
@@ -399,6 +404,10 @@ public static bool IsInitialized(object proxy)
 			{
 				return !((INHibernateProxy)proxy).HibernateLazyInitializer.IsUninitialized;
 			}
+			else if (proxy is ILazyInitializedCollection)
+			{
+				return ((ILazyInitializedCollection)proxy).WasInitialized;
+			}
 			else if (proxy is IPersistentCollection)
```

Comment:
Another todo here would be great, like `6.0 TODO: remove once IPersistentCollection derives from ILazyInitializedCollection`.